packages/frontend/src/features/catalog/hooks/useCatalog.ts

```diff
@@ -0,0 +1,37 @@
+import { type ApiCatalogResults, type ApiError } from '@lightdash/common';
+import { useQuery } from '@tanstack/react-query';
+import { lightdashApi } from '../../../api';
+import useQueryError from '../../../hooks/useQueryError';
+
+export type GetCatalogParams = {
+    projectUuid: string;
+    allTables: boolean;
+    allFields: boolean;
+};
+
+const fetchCatalog = async ({
+    projectUuid,
+    allTables,
+    allFields,
+}: GetCatalogParams) =>
+    lightdashApi<ApiCatalogResults>({
+        url: `/projects/${projectUuid}/dataCatalog?allTables=${allTables}&allFields=${allFields}`,
+        method: 'GET',
+        body: undefined,
+    });
+
+export const useCatalog = (
+    projectUuid: string,
+    allTables: boolean,
```

Comment:
I think the filters `alltables, allfields` and `search` should be like this 

```

//Initially dataResults only returns pages (allPages=true)
   const { dataResults, mutation } = useCatalog(projectUuid);
   onSearch(search) {
      mutation.mutate({search})
   } 
   onFieldTabSelected() {
       mutation.mutate({allFields=true})
   }
   onTableTabSelected() {
   //This should always return from cache, because it is the same request we load initially.
       mutation.mutate({allTables=true})
   }

```